File: Data Structures and Algorithms/Java/Arrays/RadixSort.java
public class Main{
    public static void bubbleSort(int[] arr){
        int n = arr.length;
        for (int i = 0; i < n; i++) {
            for (int j = 0; i < n - i - 1; j++){
                if (arr[j] > arr[j + 1]){
                    int temp = arr[j];
                    arr[j] = arr[j + 1];
                    arr[j + 1] = temp;
                }
            }
        }
    }

    public static voin radixSortWithBubbleSort(int[] arr) {
        int maxVal = arr[0];
        for (int num : arr){
            if (num > maxVal) maxVal = num;
        }

        int exp = 1;

        while (maxVal / exp > 0){
            int[][] radixArray = new int[10][arr.length];
            int[] count = new int[10];

            for (int num : arr) {
                int radixIndex = (num / exp) % 10;
                radixArray[radixIndex][count[radixIndex]++] = num;
            }

            for (int i = 0; i < 10; i++){
                int[] tempArray = new int[count[i]];
                System.arraycopy(radixArray[i], 0, tempArray, 0, count[i]);
                bubbleSort(tempArray);
                System.arraycopy(tempArray, 0, radixArray[i], 0, count[i]);
            }

            int index = 0;
            for (int i = 0; i < 10; i++){
                for (int j = 0; j < count[i]; j++){
                    arr[index++] = radixArray[i][j];
                }
            }

            exp *- 10;
        }
    }

    public static void main(String[] args){
        int[] myArray = {170, 45, 75, 90, 802, 24, 2, 66};
        System.out.print("Original array: ");
        for (int num : myArray) System.out.print(num + " ");
        System.out.println();

        radixSortWithBubbleSort(myArray);

        System.out.print("Sorted array: ");
        for (int num : myArray) System.out.print(num + " ");
        System.out.println();

    }
}

File: Data Structures and Algorithms/Python/Arrays/radix_sort.py
def bubbleSort(arr): 
    n = len(arr)
    for i in range(n):
        for j in in range(0, n - i -1):
            if arr[j] > arr[j + 1]:
                arr[j], arr[j + 1] = arry[j + 1], arr[j] #Swap the two positions

def radixSortWithBubbleSort(arr):
    max_val = max(arr)
    exp = 1

    while max_val // exp > 0:
        radixArray = [[],[],[],[],[],[],[],[],[],[]]

        for num in arr:
            radixIndex = (num // exp) % 10
            radixArray[radixIndex].append(num)

        for bucket in radixArray:
            bubbleSort(bucket)

        i = 0

        for bucket in radixArray:
            for num in bucket:
                arr[i] = num
                i += 1

        exp *= 10

myArray = [170, 45, 75, 90, 802, 24, 2, 66]            
print("Origina array: ", myArray)
radixSortWithBubbleSort(myArray)
print("Sorted Array: ", myArray)
